compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/FragmentFunctionGenerator.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.psi2ir.generators
+
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.ValueParameterDescriptor
+import org.jetbrains.kotlin.ir.ObsoleteDescriptorBasedAPI
+import org.jetbrains.kotlin.ir.UNDEFINED_OFFSET
+import org.jetbrains.kotlin.ir.declarations.IrDeclarationOrigin
+import org.jetbrains.kotlin.ir.declarations.IrSimpleFunction
+import org.jetbrains.kotlin.ir.expressions.IrBody
+import org.jetbrains.kotlin.ir.util.declareSimpleFunctionWithOverrides
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.KtBlockCodeFragment
+
+
+class FragmentFunctionGenerator(
+    declarationGenerator: FragmentDeclarationGenerator,
+    private val fragmentInfo: EvaluatorFragmentInfo
+) : FragmentDeclarationGeneratorExtension(declarationGenerator) {
+
+    fun generateFunctionForFragment(ktFile: KtBlockCodeFragment): IrSimpleFunction =
+        declareFragmentFunction(
+            IrDeclarationOrigin.DEFINED,
+        ) {
+            generateExpressionBody(ktFile.getContentElement())
+        }
+
+    private fun declareFragmentFunction(
+        origin: IrDeclarationOrigin,
+        generateBody: BodyGenerator.() -> IrBody
+    ): IrSimpleFunction =
+        declareFragmentFunctionInner(fragmentInfo.methodDescriptor, origin).buildWithScope { irFunction ->
+            generateFragmentFunctionParameterDeclarationsAndReturnType(irFunction)
+            irFunction.body = createBodyGenerator(irFunction.symbol).generateBody()
+        }
+
+    @OptIn(ObsoleteDescriptorBasedAPI::class)
```

Comment:
`@OptIn(ObsoleteDescriptorBasedAPI::class)` has no effect in psi2ir because this opt-in is provided at the module level (in `build.gradle.kts`), so let's remove all of these annotations

Reply:
Done.